Detections/ASimWebSession/PossibleDGAContacts.yaml

```diff
@@ -0,0 +1,104 @@
+id: 9176b18f-a946-42c6-a2f6-0f6d17cd6a8a
+name: Possible contact with a domain generated by a DGA
+description: |
+  'Identifies contacts with domains names that might have been generated by a Domain Generation Algorithm (DGA). DGAs can be used
+  by malware to generate rendezvous points that are difficult to predict in advance. This detection uses the Alexa Top 1 million domain names to build a model
+  of what normal domains look like. It uses this to identify domains that may have been randomly generated by an algorithm.
+  The triThreshold is set to 500 - increase this to report on domains that are less likely to have been randomly generated, decrease it for more likely.
+  The start time and end time look back over 6 hours of data and the dgaLengthThreshold is set to 8 - meaning domains whose length is 8 or more are reported.'
+severity: Medium
+requiredDataConnectors: []
+queryFrequency: 6h
+queryPeriod: 6h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CommandAndControl
+tags:
+  - Id: 4acd3a04-2fad-4efc-8a4b-51476594cec4
+    version: 1.0.0
+  - Schema: ASIMNetworkSession
+    SchemaVersion: 0.2.0
+relevantTechniques:
+  - T1568
+query: |
+    let triThreshold = 500;
+    let querystarttime = 6h;
+    let dgaLengthThreshold = 8;
+    // fetch the alexa top 1M domains
+    let top1M =  (externaldata (Position:int, Domain:string)   [@"http://s3-us-west-1.amazonaws.com/umbrella-static/top-1m.csv.zip"]  with (format="csv", zipPattern="*.csv"));
```

Comment:
the Alexa top 1M is very old. We should discuss with MSTIC migrating (https://blog.majestic.com/development/majestic-million-csv-daily/)

Reply:
Did you update to the link I suggested? Doesn't look so in my view? Any issue if not?

Reply:
I think Alexa is more reliable than majestic, and prefer to be as similar to parent.

Reply:
OK. I now checked. The thing is that Alexa is not available for a few years. I thought that the original link was an old copy, but it seems it is not related to Alexa at all, but rather this: http://s3-us-west-1.amazonaws.com/umbrella-static/index.html. So I am find with keeping. Just change the comment to reflect the true source.